test/jdk/tools/jlink/basic/AllModulePath.java

```diff
@@ -44,35 +37,64 @@
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.regex.Pattern;
+import java.util.spi.ToolProvider;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import java.util.spi.ToolProvider;
-
-import jdk.test.lib.compiler.CompilerUtils;
-import jdk.test.lib.process.ProcessTools;
 
 import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
-import static org.testng.Assert.*;
 
+import jdk.test.lib.compiler.CompilerUtils;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jdk.tools.jlink.internal.LinkableRuntimeImage;
+import tests.Helper;
+import tests.Result;
+
+/*
+ * @test
+ * @summary jlink test of --add-module ALL-MODULE-PATH
+ * @library ../../lib /test/lib
+ * @modules jdk.compiler
+ *          java.base/jdk.internal.jimage
+ *          jdk.jlink/jdk.tools.jlink.internal
+ *          jdk.jlink/jdk.tools.jimage
+ * @build jdk.test.lib.process.ProcessTools
+ *        jdk.test.lib.process.OutputAnalyzer
+ *        jdk.test.lib.compiler.CompilerUtils
+ * @run testng/othervm -Duser.language=en -Duser.country=US AllModulePath
```

Comment:
> Is `-Duser.language=en -Duser.country=US` needed?

I think so, since the test asserts the error message which is potentially localized.

> Please add `@bug` to this test.

Sure.